Polyhedron/demo/Polyhedron/Scene_group_item.cpp

```diff
@@ -141,70 +141,25 @@ void Scene_group_item::draw(CGAL::Three::Viewer_interface* viewer) const {
     }
     switch(child->renderingMode()) {
     case Points:
+    case ShadedPoints:
     case PointsPlusNormals:
       child->drawPoints(viewer); break;
     default: break;
     }
+    if(child->renderingMode() == Splatting)
+      child->drawSplats(viewer); break;
   }
   already_drawn = true;
 }
 
 void Scene_group_item::drawEdges(CGAL::Three::Viewer_interface* viewer) const
 {
-  if(viewer->inDrawWithNames() || already_drawn ) return;
-  Q_FOREACH(Scene_item* child, children) {
-    if(!child->visible()) continue;
-    switch(child->renderingMode()) {
-    case FlatPlusEdges:
-    case Wireframe:
-    case PointsPlusNormals:
-      child->drawEdges(viewer); break;
-    default: break;
-    }
-    switch(child->renderingMode()) {
-    case Flat:
-    case FlatPlusEdges:
-    case Gouraud:
-      child->draw(viewer); break;
-    default: break;
-    }
-    switch(child->renderingMode()) {
-    case Points:
-    case PointsPlusNormals:
-      child->drawPoints(viewer); break;
-    default: break;
-    }
-  }
-  already_drawn = true;
+  draw(viewer);
```

Comment:
I strongly disagree! If `drawEdges` and `draw` call the exact same code, and in the `FlatPlusEdges` mode, the drawing will be done twice, and in the `Writeframe` mode, the drawing will be wrong.

There may be a bug in the current logic, but the solution cannot be what you proposed.